packages/interceptors-opentelemetry/src/instrumentation.ts

```diff
@@ -43,8 +43,10 @@ async function wrapWithSpan<T>(
     span.setStatus({ code: otel.SpanStatusCode.OK });
     return ret;
   } catch (err: any) {
+    const isBenignErr = err instanceof ApplicationFailure && err.category === ApplicationFailureCategory.BENIGN;
     if (acceptableErrors === undefined || !acceptableErrors(err)) {
-      span.setStatus({ code: otel.SpanStatusCode.ERROR, message: err instanceof Error ? err.message : String(err) });
+      const statusCode = isBenignErr ? otel.SpanStatusCode.UNSET : otel.SpanStatusCode.ERROR;
+      span.setStatus({ code: statusCode, message: err instanceof Error ? err.message : String(err) });
```

Comment:
Being an `Error` is neither necessary nor sufficient to deduce that `message` will be present. Prefer this instead:

```suggestion
      span.setStatus({ code: statusCode, message: (err as Error).message ?? String(err) });
```